### PR TITLE
Add `jump_url` property to Channels

### PR DESCRIPTION
**Pull request by Pysics**
## Summary

- `TextChannel`
- `VoiceChannel`
- `StageChannel`
- `ForumChannel`
- `DMChannel`
- `GroupChannel`

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by LostLuma** [comment]
This is missing the property on `ForumChannel`, although I'm unsure this will be merged as the feature has been denied previously (see #2355).

**Comment by Vioshim** [comment]
> This is missing the property on `ForumChannel`, although I'm unsure this will be merged as the feature has been denied previously (see #2355).

To be fair, back then we didn't have discord components, this jump_url feature in channels would be very useful to redirect users through URL Buttons in views.

**Comment by Gobot1234** [comment]
This is also missing support for DMChannel is it not?

**Comment by Pysics** [comment]
For now, I have made this PR a draft.
I will look into implementing more features tomorrow.

**Comment by dolfies** [comment]
> This is missing the property on `ForumChannel`, although I'm unsure this will be merged as the feature has been denied previously (see #2355).

Copy link is an option in the client's context menu now, so it isn't a "hack" anymore.

**Comment by Pysics** [comment]
I have added more support and this PR is ready for review.

File: discord/channel.py [hunk 1]
```diff
@@ -1059,6 +1066,13 @@ def last_message(self) -> Optional[Message]:
         """
         return self._state._get_message(self.last_message_id) if self.last_message_id else None
 
+    @property
+    def jump_url(self) -> str:
+        """:class:`str`: Returns a URL that allows the client to jump to the channel.
+        .. versionadded:: 2.0
```
Comment (Rapptz) [inline discord/channel.py:1072]: These need a newline between them.

```suggestion

        .. versionadded:: 2.0
```

**Review by Rapptz — CHANGES_REQUESTED** [review]
Rather than copying this implementation everywhere in every guild channel type you probably want to do it inside abc.GuildChannel.

**Comment by Pysics** [comment]
I have not added support for `CategoryChannel`.
So, I can't add the `jump_url` property to `abc.GuildChannel`.

**Comment by Rapptz** [comment]
> I have not added support for `CategoryChannel`. So, I can't add the `jump_url` property to `abc.GuildChannel`.

I'd rather it be in all channels than duplicate all of this. Plus, this would play better with type checkers.

**Comment by Pysics** [comment]
@Rapptz I have made the requested changes.

**Comment by Rapptz** [comment]
This is failing type checking.

You don't want to add this jump_url property to the abc.PrivateChannel protocol. That one will have to be done manually to the things that meet the protocol since it isn't an actual runtime base class.

**Review by Rapptz — CHANGES_REQUESTED** [review]
Needs to fix type checking concern.

**Review by Rapptz — APPROVED** [review]
Thanks. LGTM.

**Pull request merged**